### ISSUE TITLE
[a11y]: List Page_Screen reader announces the entire table content after pagination

### ISSUE THREAD

**Issue by anamhussain94** [carbon-design-system/carbon#21055]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.96.0

### Automated testing tool and ruleset

WCAG Mapping: 4.1.2

### Assistive technology

MacOS voiceover

### Description

When a user activates the next or previous button in pagination, the table updates with new content. However, screen readers automatically read out the entire updated table, makes it harder for screen reader users to stay oriented.

Avoid announcing the entire table. Instead, use a brief message like “Items 21–30 of 88” to confirm the update, and ensure it is announced by screen readers.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-datatable-basic--overview&addons=1&panelRight=0&selectedKind=Accordion&selectedStory=Default&stories=1

### Steps to reproduce

When a user activates the next or previous button in pagination, the table updates with new content. However, screen readers automatically read out the entire updated table, makes it harder for screen reader users to stay oriented.

Avoid announcing the entire table. Instead, use a brief message like “Items 21–30 of 88” to confirm the update, and ensure it is announced by screen readers.

Steps to reproduce

Navigate to carbon/design [DataTable](https://react.carbondesignsystem.com/?path=/docs/components-datatable-dynamic--overview)

tbody has **aria-live="polite"** property due to which screenreader reads the entire table content when user clicks on the pagination next and previous buttons .

<img width="1471" height="337" alt="Image" src="https://github.com/user-attachments/assets/3b0565ca-1abf-474d-acbc-808fab772c2e" />

On click of pagination button , screenreader should announce the number of items.
https://github.com/carbon-design-system/carbon/issues/21054 
Defect for the same is opened.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by 2nikhiltom** [comment]
dup of https://github.com/carbon-design-system/carbon/issues/21054
